src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/ReInviteUserProfileIntTest.java

```diff
@@ -179,4 +179,22 @@ void sld_rtn_429_when_usr_reinvited_successfully_and_again_reinvited_in_1_hour_w
                 .concat("%s minutes ago. Please try after some time"), resendInterval));
     }
 
+    @Test
+    void should_update_upId_with_idamId_incase_diff_when_user_reinvited() throws Exception {
+
+        updateLastUpdatedTimestamp(userProfile.getIdamId());
+
+        UserProfileCreationData data = buildCreateUserProfileData(true);
+        data.setEmail(pendingUserRequest.getEmail());
+        searchUserProfileSyncWireMock(HttpStatus.OK);
+        UserProfileCreationResponse reInvitedUserResponse = (UserProfileCreationResponse) createUser(data, CREATED,
+                UserProfileCreationResponse.class);
+
+        //getting from DB
+        Optional<UserProfile> persistedUserProfile = userProfileRepository.findByEmail(pendingUserRequest
+                .getEmail().toLowerCase());
+        userProfile = persistedUserProfile.get();
+
+        assertThat(reInvitedUserResponse.getIdamId()).isEqualTo(userProfile.getIdamId());
```

Comment:
add another assertion to verify updated new idam id is not same as old idam id

Reply:
Thanks @rajkatla-hmcts been added !!!